spec/inject_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'inject'
+
+describe Array do
+  describe 'new_inject_block' do
+    it 'should use first element of array as default argument' do
+      expect([3,4,5].new_inject_block {|a,b| a + b}).to eq 12
+    end
+
+    it 'should use first element of array as default argument' do
+      expect([1,2,3].new_inject_block {|a,b| a + b}).to eq 6
+    end
+
+    it 'should accept an argument as a starting value' do
+      expect([3,4,5].new_inject_block(5) {|a,b| a + b}).to eq 17
```

Comment:
Space missing after comma.<br>Space between { and | missing.<br>Space missing inside }.